### DISCUSSION TITLE
ImageResize plugin support

### DISCUSSION THREAD

**Discussion by ItagMichael** [craftcms/ckeditor#94] [Ideas] [answered]
Currently, the CKEditor build that ships with this package doesn't have the [ImageResize](https://ckeditor.com/docs/ckeditor5/latest/api/module_image_imageresize-ImageResize.html) plugin included. This plugin allows for manual resize handles as well as scaling images to preset widths.

Obviously, this should not be a replacement for image transforms since we're just scaling images in CSS, but resizing images works well alongside the functionality of [ImageStyle](https://ckeditor.com/docs/ckeditor5/latest/features/images/images-styles.html) which is included by default since it appears in the same toolbar.

We may use ImageStyle to replicate (sparingly) floating images sometimes for clients and it is much less flexible without ImageResize.

**Comment by brandonkelly** [comment, marked as answer]
We just released CKEditor 3.4.0 with the ability to change the selected transform for inserted images:

<img width="770" alt="Screenshot of an inserted image in CKEditor, with a floating toolbar that begins with a transform selection dropdown, with “Medium” currently selected." src="https://github.com/craftcms/ckeditor/assets/47792/1fbcd466-84ea-422b-8651-36a9187cc153">

It’s limited to the list of Available Transforms your field is configured with, as well as an “Original” option.

(We chose to go this route instead of pulling in the ImageResize plugin, as ImageResize only touches the `<img>`’s `width` attribute; it has no control over the loaded image size. Using transforms means the image will be served with the correct size in the first place.)

**Comment by rtrudel** [comment]
How to make this popover show ? It don't seem to appear out of the box.

**Reply by brandonkelly** [reply]
Are you on 3.4.0? Have you made any configuration changes that could be suppressing it? Any JS errors?

**Reply by rtrudel** [reply]
This has been solved automatically after a Craft update I guess, I just noticed. We'll never know  `¯\_(ツ)_/¯`

**Comment by mark-chief** [comment]
Sad the hear the imageResize plugin wasn't shipped, it's a super helpful tool for publishers to make small adjustments to images, quickly, where the image doesn't actually resize its not really an issue. Is there a way to install this manually instead? Thanks

**Reply by gjhead** [reply]
I agree.  It's not the most ideal way to resize images but it does have a use case - especially when it comes to migrating large sites over to Craft 5.  While I could understand keeping this feature off by default in Craft, I would prefer the option to turn it on manually until a better solution can be built in the CMS and editors trained on how to properly use it.  @brandonkelly

**Comment by mark-chief** [comment]
In the meantime how can i get this dropdown to appear over my image? I have set a transform but im not seeing it.. craft 4.5.11.1 cke 3.6.0
thanks
<img width="656" alt="Screenshot 2023-12-11 at 17 29 43" src="https://github.com/craftcms/ckeditor/assets/18699249/77d8edb6-7e06-4f03-a868-9239581b1b03">

**Reply by i-just** [reply]
@mark-chief by the looks of it, you already have something like this in your CKEditor config:
```
"image": {
  "toolbar": [
    "imageStyle:alignLeft",
    "imageStyle:alignCenter",
    "imageStyle:alignRight",
    "toggleImageCaption",
    "imageTextAlternative"
  ]
}
```
if you add `"transformImage",` to that list (and you have transforms defined in the Control Panel), the transforms dropdown should appear.

**Reply by mark-chief** [reply]
Hi @i-just , Thanks.. I have the transform set up and the cke config, but it's not actually doing anything when I select the transform from the dropdown and then save..

<img width="408" alt="Screenshot 2024-01-05 at 10 12 24" src="https://github.com/craftcms/ckeditor/assets/18699249/143943d6-e8bb-4857-b413-e78a1762601e">

**Reply by brandonkelly** [reply]
@mark-chief We just released CKEditor 3.8.0 which now updates the image’s `width` and `height` attributes when a new transform is selected (#165).

**Comment by jamesevans1808** [comment]
I'm trying to set up the ck config to allow for the use of image transforms as mentioned above, but I get the warning in the console that the item "transformImage" for toolbar is unavailable (toolbarview-item-unavailable).

Could anyone please help? It'd be great to be able to resize images within the ckeditor. Thanks

EDIT: running Craft CMS 4.13.5, CKEditor 3.10.0, PHP 8.2.17

**Reply by i-just** [reply]
Did you select any available transforms for your CKEditor field(s) (control panel → Settings → Fields → _your CKEditor field_)?

<img width="578" alt="Screenshot 2024-12-05 at 11 16 43" src="https://github.com/user-attachments/assets/b7ec73e0-2bbb-4a8f-b333-0cde24a63351">

**Reply by jamesevans1808** [reply]
I'd missed that part, I've done that and it's resolved my issue. Whoops! Thanks a bunch, @i-just !
